frappe/templates/pages/integrations/razorpay_checkout.py

```diff
@@ -18,6 +18,8 @@ def get_context(context):
 
 	try:
 		doc = frappe.get_doc("Integration Request", frappe.form_dict['token'])
+		if doc.status != 'Queued':
+			raise Exception
```

Comment:
I think instead of raising we can use `frappe.throw()` with an exception that tells the user what exactly the issue is.

Reply:
`frappe.throw` does not seem to work inside `get_context`. It is logged as a server exception only.

Furthermore, the exception is already handled by this block below which redirects and informs the user that the token is "invalid". "Invalid" in this case happens when the _Integration Request_ does not exists. I thought probably an existing but "used" token could be considered "invalid" as well. Please let me know if I should update the message to be more specific or whether the existing catch-all is sufficient.

Reply:
No I do think something more specific is needed in this case. Let me have a look.

Reply:
Would you prefer something like the following?

```py
...
class TokenAlreadyUsedError(Exception):
	pass

def get_context(context):
	...
	try:
		doc = frappe.get_doc("Integration Request", ...
		if doc.status != 'Queued':
			raise TokenAlreadyUsedError
		...
	except frappe.exceptions.DoesNotExistError as e:
		frappe.redirect_to_message(_('Invalid Token'),
			_('Seems token you are using is invalid!'),
			http_status_code=400, indicator_color='red')
		...
	except TokenAlreadyUsedError as e:
		frappe.redirect_to_message(_('Token Already Used'),
			_('It seems you have already paid through this link!'),
			http_status_code=400, indicator_color='red')
		raise frappe.Redirect
```

Reply:
yeah works

Reply:
Made changes.